### PR TITLE
feat(slack): Add check function for no-bot channel.

### LINKED ISSUE #123: Prompt to add bot to channels where slash commands are used

### PR DESCRIPTION
**Pull request by Ayush0Chaudhary**
Fixes #123

### CODE REVIEW

File: bot/slack/runner.py [hunk 1]
```diff
@@ -20,9 +23,10 @@ class Runner(SlackBotBase):
 
     logger: Logger
 
-    def __init__(self, logger: Logger):
+    def __init__(self, token: str, logger: Logger, r):
```
Comment (BURG3R5) [inline bot/slack/runner.py:26]: Remove this `, r`

File: bot/slack/__init__.py [hunk 2]
```diff
@@ -18,6 +18,6 @@ class SlackBot(Messenger, Runner):
     Specifics are delegated to parent classes `Messenger` and `Runner`.
     """
 
-    def __init__(self, token: str, logger: Logger):
+    def __init__(self, token: str, logger: Logger, bot_id: str):
         Messenger.__init__(self, token)
-        Runner.__init__(self, logger)
+        Runner.__init__(self, token, logger, bot_id)
```
Comment (BURG3R5) [inline bot/slack/__init__.py:23]: `bot_id` is not required

File: bot/slack/runner.py [hunk 3]
```diff
@@ -274,3 +279,34 @@ def mini_help_response(text: str) -> dict[str, Any]:
                 },
             ],
         }
+
+    def check_bot_in_channel(
+        self,
+        current_channel: str,
+    ) -> dict[str, Any]:
+        subscriptions = self.storage.get_subscriptions(current=current_channel)
```
Comment (BURG3R5) [inline bot/slack/runner.py:287]: ??? Did you even test this?

Comment (BURG3R5) [reply bot/slack/runner.py:287]: `current` is NOT an argument of `get_subscriptions`; use `channel=`

File: bot/slack/runner.py [hunk 4]
```diff
@@ -274,3 +279,34 @@ def mini_help_response(text: str) -> dict[str, Any]:
...
+                                "type":
+                                "mrkdwn",
+                                "text":
+                                "I subscribed this repo to the channel,\n but I am not in this channel, \nIf you need me to send you updates, invite me in this channel first."
+                            }
+                        }]
+                    }
+        except SlackApiError as E:
+            capture_message(
+                f"SlackApiError {E} Failed to fetch conversation info for #{current_channel}"
```
Comment (BURG3R5) [inline bot/slack/runner.py:311]: `current_channel` already has the `"#"` in it, no need to add another

**Review by BURG3R5 — CHANGES_REQUESTED** [review]
There's a lot of problems with this. First fix the things I mentioned in the comments, then I'll tell you what else is wrong

**Comment by Ayush0Chaudhary** [comment]
will test it by tomorrow eod

**Pull request closed without merging**